lib/clerk/authenticate_context.rb

```diff
@@ -0,0 +1,189 @@
+        # Expose the url of the request that this parameter object was created from as a URI object.
+        attr_reader :clerk_url
+
+        ##
+        # Expose properties that does not require validations or complex logic to retrieve
+        # values by delegating them to the cookies or headers variables.
+        def_delegators :@cookies, :session_token_in_cookie, :client_uat
+        def_delegators :@headers, :session_token_in_header, :sec_fetch_dest
+
+        ##
+        # Creates a new parameter object using Rack::Request and Clerk::Config objects.
+        def initialize(request, config)
+            @clerk_url = URI.parse(request.url)
+            @config = config
+
+            @cookies = OpenStruct.new({
+                session_token_in_cookie: request.cookies[SESSION_COOKIE],
+                client_uat: request.cookies[CLIENT_UAT_COOKIE],
+                handshake_token: request.cookies[HANDSHAKE_COOKIE],
+                dev_browser: request.cookies[DEV_BROWSER_COOKIE]
+            })
+
+            header_token = request.env[AUTHORIZATION_HEADER] || ''
+            @headers = OpenStruct.new({
+                session_token_in_header: header_token.gsub(/bearer/i, '').strip,
+                sec_fetch_dest: request.env[SEC_FETCH_DEST_HEADER],
+                accept: request.env[ACCEPT_HEADER] || "",
+                origin: request.env[ORIGIN_HEADER] || "",
+                host: request.host,
+                port: request.port
+            })
+        end
+
+        ##
+        # The following properties are part of the props supported in all the AuthenticateContext
+        # objects across all of our SDKs (eg JS, Go)
+        def secret_key
+            @config.api_key || ""
+        end
+
+        def publishable_key
+            @config.publishable_key || ""
+        end
+
+        def domain
+            # TODO(dimkl): Add multi-domain support
+        end
+
+        def is_satellite?
+            # TODO(dimkl): Add multi-domain support
+            false
+        end
+
+        def proxy_url
+            # TODO(dimkl): Add multi-domain support
+        end
+
+        def handshake_token
+            @handshake_token ||= retrieve_from_query_string(@clerk_url, HANDSHAKE_COOKIE) || @cookies.handshake_token || ""
+        end
+
+        def clerk_synced?
+            # TODO(dimkl): Add multi-domain support
+            false
+        end
+
+        def clerk_redirect_url
+             # TODO(dimkl): Add multi-domain support
+             ""
+        end
+
+        ##
+        # Helper methods supported only by Ruby SDK to reduce code duplication and improve readability
+        #
+
+        def dev_browser
+            @dev_browser ||= retrieve_from_query_string(@clerk_url, DEV_BROWSER_COOKIE) || @cookies.dev_browser || ""
+        end
+
+        # The frontend_api returned is without protocol prefix
+        def frontend_api
+            return "" if !publishable_key || !valid_publishable_key?(publishable_key)
```

Comment:
:upside_down_face: Perhaps the `valid_publishable_key?` method can also check that the key is not blank, eliminating the need to add this OR clause here.

Or we can just do 
```ruby
return "" if !valid_publishable_key?(publishable_key.to_s)
```